hw/misc/hermes.c

```diff
@@ -287,19 +294,20 @@ static void hermes_dma_thread(HermesState *hermes,
 static void *hermes_h2c_dma_thread(void *opaque)
 {
     HermesState *hermes = opaque;
-    hermes_dma_thread(hermes, &hermes->bar2.h2c);
+    for(size_t i = 0; i < H2C_CHANNELS; i++)
+        hermes_dma_thread(hermes, &hermes->bar2.h2c[i]);
```

Comment:
We don't want the loop here. `hermes_h2c_dma_thread()` is the entry point for the threads created on `hermes_bar2_init_dir_reg()`, so we already have `H2C_CHANNELS` threads running.

The "problem" then is knowing which `hermes_bar_dir_reg` to use when calling `hermes_dma_thread()`. One solution might be to use the dir_reg as the argument when creating the thread. Then you can use the `container_of` macro to get `hermes`.